platform/android/app/src/androidTest/kotlin/coop/polypoly/polypod/FeatureSmokeTest.kt

```diff
@@ -0,0 +1,69 @@
+package coop.polypoly.polypod
+
+import androidx.compose.ui.test.SemanticsNodeInteraction
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.hasTestTag
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import androidx.compose.ui.test.onAllNodesWithTag
+import androidx.compose.ui.test.performClick
+import androidx.compose.ui.test.performScrollTo
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.doesNotExist
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class FeatureSmokeTest {
+
+    @get:Rule
+    val composeTestRule = createAndroidComposeRule(MainActivity::class.java)
+
+    private fun nodeIsDisplayed(nodeInteraction: SemanticsNodeInteraction): Boolean { // ktlint-disable max-line-length
+        return nodeInteraction.runCatching {
+            nodeInteraction.assertIsDisplayed()
+        }.isSuccess
+    }
+
+    private fun homeScreen(): SemanticsNodeInteraction {
+        return composeTestRule.onNode(hasTestTag("HomeScreen"))
+    }
+
+    private fun pressCloseUntilHomeScreenIsDisplayed() {
+        var retryCount = 0
+        while (!nodeIsDisplayed(homeScreen()) && retryCount < 10) {
```

Comment:
I think you can configure the app to not show the onboarding by configuring preferences -> https://github.com/polypoly-eu/polyPod/blob/a9a5efbfb782ad3f62a861052826753e60f0b9c1/platform/android/app/src/androidTest/kotlin/coop/polypoly/polypod/OnboardingTest.kt#L38

Reply:
True. I will leave it like this though.

Reply:
It is your choice, 
but configuring the app to not show the onboarding has the potential to remove the whole `while loop`.

Reply:
Still needed down below, when I try to close the feature.

Reply:
Fair enough. The improvement can be done in a separate PR, as we should definitely try to start the app directly at the Home Screen to avoid any side effects from other flows.

Reply:
🤔 Good point. I will add that!

Reply:
@GhenadiePusca Just tried to configure it like you suggested. It does not work. Note, I am not using espresso to show the homescreen. Unfortunately espresso does not have good support for jetpack compose. I needed to use a ui testing frameworks specifically for compose.